sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -527,7 +530,7 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
     }
   }
 
-  object ResolveGroupingAnalytics extends Rule[LogicalPlan] {
+  class ResolveGroupingAnalytics(catalog: SessionCatalog) extends Rule[LogicalPlan] {
```

Comment:
it doesn't appear that this rule uses this newly provided `catalog` anywhere?

Reply:
yea we can remove the change.

Reply:
Done